pkg/secrets_reflector/secrets_reflector.go

```diff
@@ -0,0 +1,153 @@
+		return nil, err
+	}
+	key64, ok := kubeSecret.Data["tls.key"]
+	if !ok {
+		err := fmt.Errorf("invalid secret: %s/%s does not contain tls.key field", kubeSecret.Namespace, kubeSecret.Name)
+		return nil, err
+	}
+	// base64 decode tls.crt and tls.pem fields
+	crt, err := base64.StdEncoding.DecodeString(string(crt64))
+	if err != nil {
+		err = fmt.Errorf("invalid secret: error decoding %s/%s tls.crt field: %v", kubeSecret.Namespace, kubeSecret.Name, err)
+		return nil, err
+	}
+	key, err := base64.StdEncoding.DecodeString(string(key64))
+	if err != nil {
+		err = fmt.Errorf("invalid secret: error decoding %s/%s tls.key field: %v", kubeSecret.Namespace, kubeSecret.Name, err)
+		return nil, err
+	}
+	// concatenate crt and private key and put the result in an
+	// DCOS Secret
+	var sb strings.Builder
+	sb.Write(crt)
+	sb.Write(key)
+
+	dcosSecret := &dcos.SecretsV1Secret{
+		Value: sb.String(),
+	}
+	return dcosSecret, nil
+}
+
+// reflect checks if kubernetes secret was updated by verifying if the
+// MD5 hash of certificate and private key changed and if required
+// proceeds to re-create a DCOS secret to match it.
+func (s *secretsReflector) reflect(kubeSecret *corev1.Secret, dcosSecret *dcos.SecretsV1Secret) error {
+	hashRaw := md5.Sum([]byte(dcosSecret.Value))
+	hash := fmt.Sprintf("%x", hashRaw)
+	expectedHash := kubeSecret.Annotations[constants.DklbSecretAnnotationKey]
+	// hash did not change so we can exit now
+	if len(hash) > 0 && len(expectedHash) > 0 && hash == expectedHash {
+		s.logger.Infof("no changes to secret %s/%s detected, skipping update", kubeSecret.Namespace, kubeSecret.Name)
+		return nil
+	}
+	// generate the DCOS secret name
+	dcosSecretName := fmt.Sprintf("%s__%s__%s", s.kubernetesClusterName, kubeSecret.Namespace, kubeSecret.Name)
+
+	ctx, fn := context.WithTimeout(context.Background(), defaultTimeout)
+	defer fn()
+	// delete the secret from DCOS secret store and ignore any
+	// errors (ex: AlreadyExists or NotFound)
+	s.logger.Infof("(re)creating DCOS secret %s", dcosSecretName)
+	s.dcosSecretsClient.DeleteSecret(ctx, defaultSecretStore, dcosSecretName)
+	// create the DCOS secret
+	_, err := s.dcosSecretsClient.CreateSecret(ctx, defaultSecretStore, dcosSecretName, *dcosSecret)
+	if err != nil {
+		s.logger.Errorf("failed to (re)create DCOS secret %s: %s", dcosSecretName, err)
+		return err
+	}
+	// add the hash annotation to the kubernetes secret and update it
+	if kubeSecret.Annotations == nil {
+		kubeSecret.Annotations = make(map[string]string)
+	}
+	kubeSecret.Annotations[constants.DklbSecretAnnotationKey] = hash
+	s.logger.Infof("detected changes to secret %s/%s: updating annotation", kubeSecret.Namespace, kubeSecret.Name)
+	if _, err := s.kubeClient.CoreV1().Secrets(kubeSecret.Namespace).Update(kubeSecret); err != nil {
+		s.logger.Errorf("failed to update secret %s/%s: %s", kubeSecret.Namespace, kubeSecret.Name, err)
```

Comment:
```suggestion
		s.logger.Errorf("failed to update secret \"%s/%s\": %v", kubeSecret.Namespace, kubeSecret.Name, err)
```

Reply:
nit on suggestion: `failed to update Kubernetes secret...`